jplag.frontend.R/src/main/java/de/jplag/R/RCombinedBaseListener.java

```diff
@@ -0,0 +1,342 @@
+package de.jplag.R;
+
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.tree.ErrorNode;
+import org.antlr.v4.runtime.tree.TerminalNode;
+
+import de.jplag.R.grammar.*;
+
+/**
+ * Empty base implementation for {@link RListener} and {@link RFilterListener}.
+ */
+public class RCombinedBaseListener implements RListener, RFilterListener {
```

Comment:
It is a dummy class full of empty methods, so that in "real" implementations, empty methods do not need to be reimplemented and so the code is more readable, see [above](#discussion_r901315237).

Reply:
Let's make it abstract, as it should not be instantiated.